ddl/table.go

```diff
@@ -474,6 +479,33 @@ func onTruncateTable(d *ddlCtx, t *meta.Meta, job *model.Job) (ver int64, _ erro
 		}
 	}
 
+	newIDs := make([]int64, 0, len(oldPartitionIDs))
+	if d.infoHandle != nil && d.infoHandle.IsValid() {
+		is := d.infoHandle.Get()
+
+		bundles := make([]*placement.Bundle, 0, len(oldPartitionIDs)+1)
+		if oldBundle, ok := is.BundleByName(placement.GroupID(tableID)); ok {
+			bundles = append(bundles, placement.BuildPlacementTruncateBundle(oldBundle, newTableID))
+		}
+
+		if pi := tblInfo.GetPartitionInfo(); pi != nil {
+			newDefs := pi.Definitions
+			for i := range oldPartitionIDs {
+				if oldBundle, ok := is.BundleByName(placement.GroupID(oldPartitionIDs[i])); ok {
+					newID := newDefs[i].ID
+					newIDs = append(newIDs, newID)
+					bundles = append(bundles, placement.BuildPlacementTruncateBundle(oldBundle, newID))
+				}
+			}
+		}
```

Comment:
The placement rules shouldn't be removed from PD immediately. They can only be removed in the GC progress.

Reply:
Yes, it is truncation, or renaming. Maybe should change `BuildPlacementTruncateBundle` to `BuildPlacementBundleWithNewName` or so..